Source/WebCore/dom/DocumentFullscreen.cpp

```diff
@@ -24,40 +24,790 @@
  */
 
 #include "config.h"
+#include "DocumentFullscreen.h"
 
 #if ENABLE(FULLSCREEN_API)
-#include "DocumentFullscreen.h"
 
-#include "Element.h"
+#include "Chrome.h"
+#include "ChromeClient.h"
+#include "ElementInlines.h"
 #include "EventLoop.h"
+#include "EventNames.h"
+#include "HTMLDialogElement.h"
+#include "HTMLIFrameElement.h"
+#include "HTMLMediaElement.h"
 #include "JSDOMPromiseDeferred.h"
+#include "LocalDOMWindow.h"
+#include "LocalFrame.h"
+#include "Logging.h"
+#include "Page.h"
+#include "PseudoClassChangeInvalidation.h"
+#include "QualifiedName.h"
+#include "Quirks.h"
+#include "RenderBlock.h"
+#include "Settings.h"
+#include <wtf/LoggerHelper.h>
 
 namespace WebCore {
 
-// https://fullscreen.spec.whatwg.org/#exit-fullscreen
+using namespace HTMLNames;
+
+#pragma mark Document+Fullscreen.idl methods:
+
 void DocumentFullscreen::exitFullscreen(Document& document, RefPtr<DeferredPromise>&& promise)
 {
-    if (!document.isFullyActive() || !document.fullscreenManager().fullscreenElement()) {
+    if (!document.isFullyActive() || !document.fullscreen().fullscreenElement()) {
         promise->reject(Exception { TypeError, "Not in fullscreen"_s });
         return;
     }
-    document.fullscreenManager().exitFullscreen(WTFMove(promise));
+    document.fullscreen().exitFullscreen(WTFMove(promise));
 }
 
 void DocumentFullscreen::webkitExitFullscreen(Document& document)
 {
-    if (document.fullscreenManager().fullscreenElement())
-        document.fullscreenManager().exitFullscreen(nullptr);
+    if (document.fullscreen().fullscreenElement())
+        document.fullscreen().exitFullscreen(nullptr);
 }
 
 // https://fullscreen.spec.whatwg.org/#dom-document-fullscreenenabled
 bool DocumentFullscreen::fullscreenEnabled(Document& document)
 {
     if (!document.isFullyActive())
         return false;
-    return document.fullscreenManager().isFullscreenEnabled();
+    return document.fullscreen().isFullscreenEnabled();
+}
```

Comment:
I'm also tempted to say these should move to Document.cpp, but no strong opinions here.